website/static/js/filepage/index.js

```diff
@@ -87,9 +87,35 @@ var FileViewPage = {
         self.file = self.context.file;
         self.node = self.context.node;
         self.editorMeta = self.context.editor;
-        //Force canEdit into a bool
-        self.canEdit = m.prop(!!self.context.currentUser.canEdit);
-
+        self.file.checkout_user = null;
+        self.request_done = false;
+        self.isCheckout_user = function() {
+            $.ajax({
+                method: 'get',
+                url: window.contextVars.apiV2Prefix + 'files' + self.file.path + '/',
+                beforeSend: $osf.setXHRAuthorization
+            }).done(function(resp) {
+                self.request_done = true;
+                self.file.checkout_user = resp.data.relationships.checkout.links.related ? ((resp.data.relationships.checkout.links.related.href).split('users/')[1]).replace('/', ''): null;
+                if ((self.file.checkout_user) && (self.file.checkout_user !== self.context.currentUser.id)) {
+                    m.render(document.getElementById('alertBar'), m('.alert.alert-warning[role="alert"]', m.trust('<strong>File is checked-out.</strong> This file has been checked-out by a <a href="/' +
+                        self.file.checkout_user +'"> collaborator</a>. It needs to be checked back in before any changes can be made. </div>')));
+                }
+            }).fail(function(resp) {
+            });
+        };
+        if (self.file.provider === 'osfstorage'){
+            self.canEdit = function() {
+                return ((!self.file.checkout_user) || (self.file.checkout_user === self.context.currentUser.id)) ? (self.context.currentUser.canEdit ? true : false) : false;
```

Comment:
For `(self.context.currentUser.canEdit ? true : false)`, isn't `self.context.currentUser.canEdit` either `true` or `false` already?